Enable use as argument matcher

stephenprater opened the pull request:
Let's you use the matcher class as an argument matcher with the "object_matching_schema" method.

It also outputs some helpful information in the diff output, rather than just returning false and saying it should have matched.

rloomba commented on lib/json-schema-rspec/matchers/json_schema_matcher.rb:
```diff
@@ -17,13 +17,7 @@ def matches?(actual)
-            "  RSpec.configure do |config|",
-            "    config.json_schemas[:my_remote_schema] = 'path/to/schema'",
-            "    config.json_schemas[:my_inline_schema] = '{\"json\": \"schema\"}'",
-            "  end"]
```
Do you you think there's value in keeping this help info in the error text?

stephenprater replied:
Well... no.  I mean when I get this message, it's generally not because I didn't put ANY schemas into my `spec_helper` but because I was either wrong about the name, or forgot the particular schema a tried to load.

That said - I didn't actually mean to commit this - I changed it because I didn't find it helpful and forgot to change it back.

rloomba replied:
The README has pretty explicit instructions on how to properly set things out. I think it's fine to remove this.

rloomba commented:
thanks @stephenprater for the PR! Do  you mind adding an argument matcher example in the README?

stephenprater commented:
So I discovered, quite by accident, that the underlying library doesn't care if you pass it hashes or strings to verify against the schema, which is pretty cool.

I renamed the matcher method and added a "warning" to the readme about that behavior.

rloomba commented:
Great, thanks for adding that!
